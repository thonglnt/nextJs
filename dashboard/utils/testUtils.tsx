import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
