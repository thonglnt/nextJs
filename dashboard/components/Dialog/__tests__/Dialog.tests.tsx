// Utils
import { render } from '@utils/testUtils'

// Components
import AlertDialog, { IAlertDialog } from '..'

describe('Dialog component', () => {
  const mockProps: IAlertDialog = {
    isOpen: true,
    onClose: jest.fn(),
    title: 'Dialog Title',
    content: 'DiaLog Content',
    cancelTitle: 'Cancel',
    confirmTitle: 'Submit',
  }

  it('should render Dialog component', () => {
    const { container } = render(<AlertDialog {...mockProps} />)

    expect(container).toMatchSnapshot()
  })
})
