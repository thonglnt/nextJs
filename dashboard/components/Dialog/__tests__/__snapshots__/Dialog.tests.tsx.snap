// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component should render Dialog component 1`] = `
<div
  aria-hidden="true"
/>
`;
